Lib9c/Action/MintAssets.cs

```diff
@@ -0,0 +1,140 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using Bencodex.Types;
+using Lib9c;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Libplanet.Common;
+using Libplanet.Crypto;
+using Libplanet.Types.Assets;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+
+namespace Nekoyume.Action
+{
+    [ActionType(TypeIdentifier)]
+    public class MintAssets : ActionBase
+    {
+        public const string TypeIdentifier = "mint_assets";
+        public override IValue PlainValue =>
+            new Dictionary(
+                new[]
+                {
+                    new KeyValuePair<IKey, IValue>((Text)"type_id", (Text)TypeIdentifier),
+                    new KeyValuePair<IKey, IValue>(
+                        (Text)"values",
+                        MintSpecs is { }
+                            ? new List(MintSpecs.Select(s => s.Serialize()))
+                            : Null.Value
+                    )
+                }
+            );
+
+        public MintAssets()
+        {
+        }
+
+        public MintAssets(IEnumerable<MintSpec> specs)
+        {
+            MintSpecs = specs.ToList();
+        }
+
+        public override IAccount Execute(IActionContext context)
+        {
+            if (MintSpecs is null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            CheckPermission(context);
+            IAccount state = context.PreviousState;
+
+            foreach (var (recipient, assets, items) in MintSpecs)
+            {
+                if (assets is { } assetsNotNull)
+                {
+                    state = state.MintAsset(context, recipient, assetsNotNull);
+                }
+
+                if (items is { } itemsNotNull)
+                {
+                    Address inventoryAddr = recipient.Derive(SerializeKeys.LegacyInventoryKey);
+                    Inventory inventory = state.GetInventory(inventoryAddr);
+                    MaterialItemSheet itemSheet = state.GetSheet<MaterialItemSheet>();
+                    if (itemSheet is null || itemSheet.OrderedList is null)
+                    {
+                        throw new InvalidOperationException();
+                    }
+
+                    foreach (MaterialItemSheet.Row row in itemSheet.OrderedList)
+                    {
+                        if (row.ItemId.Equals(items.Id))
```

Comment:
```suggestion
                   var row = itemSheet[items.Id]
```

Reply:
`items.Id` is `HashDigest<SHA256>` and it doesn't seem to be used for lookup like as `itemSheet[items.Id]` since `itemSheet[]` expects only `int`.